src/views/Governance/components/DelegateForm.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { Fragment, useState, useCallback } from 'react';
+import { Spacer, Button, Input } from "react-neu";
+import Split from "components/Split";
+import { validateAddress } from 'utils';
+
+interface DelegateFormProps {
+  isStaked: boolean,
+  onDelegateUnstaked: (delegatee: string) => void,
+  onDelegateStaked: (delegatee: string) => void,
+  onRemoveDelegation: () => void
+}
+
+const DelegateForm: React.FC<DelegateFormProps> = ({
+  isStaked,
+  onDelegateStaked,
+  onDelegateUnstaked,
+  onRemoveDelegation
+}) => {
+  const [delegatee, setDelegatee] = useState('');
+
+  const handleOnDelegateStaked = useCallback(
+    () => onDelegateStaked(delegatee),
+    [onDelegateStaked, delegatee]
+  );
+
+  const handleOnDelegateUnstaked = useCallback(
+    () => onDelegateUnstaked(delegatee),
+    [onDelegateUnstaked, delegatee]
+  );
+
+  const onChange = (e: any): void => setDelegatee(e.target.value);
+
+  return (
+    <Fragment>
+      <Input onChange={onChange} placeholder="Delegate to..."></Input>
+      <Spacer />
+      <Split>
+        <Button full
+          text="Delegate"
+          variant="tertiary" 
+          onClick={isStaked ? handleOnDelegateStaked : handleOnDelegateUnstaked}
+          disabled={!validateAddress(delegatee)}
+        />
+        <Button full text="Remove Delegation" variant="tertiary" onClick={onRemoveDelegation} />
```

Comment:
we need this button to be disabled if the user has no delegations to undelegate from, you can add checks to see if the user is delegating on either contracts, using the `delegates` function with param of the current account connected, in case it returns the same account there would be nothing to undelegate from

Reply:
I have now disabled the button if didDelegate returns false, but didDelegate does not cover if you are staked. This is because the YAMETH pool contract does not have a ```delegates``` function. Suggestions? Should I turn in what I have currently?

Reply:
Fixed. Remove Delegation button now checks the ```didDelegate``` function before being enabled.